src/react/components/common/keyboardManager/keyboardManager.test.tsx

```diff
@@ -45,132 +47,179 @@ describe("Keyboard Manager Component", () => {
+                altKey: false,
+                key: "F1",
+            });
 
         window.dispatchEvent(keyboardEvent);
 
-        const registrationManagerMock = KeyboardRegistrationManager as jest.Mocked<typeof KeyboardRegistrationManager>;
-        expect(registrationManagerMock.prototype.invokeHandlers).toBeCalledWith(
-            KeyEventType.KeyDown, "F1", keyboardEvent);
+        expect(registrationManagerMock.prototype.invokeHandlers)
+            .toBeCalledWith(KeyEventType.KeyDown, "F1", keyboardEvent);
     });
 
     it("listens for keyup events and invokes handlers", () => {
         const keyboardEvent = new KeyboardEvent(
             KeyEventType.KeyUp, {
-            ctrlKey: false,
-            altKey: false,
-            key: "F1",
-        });
+                ctrlKey: false,
+                altKey: false,
+                key: "F1",
+            });
 
         window.dispatchEvent(keyboardEvent);
 
-        const registrationManagerMock = KeyboardRegistrationManager as jest.Mocked<typeof KeyboardRegistrationManager>;
-        expect(registrationManagerMock.prototype.invokeHandlers).toBeCalledWith(
-            KeyEventType.KeyUp, "F1", keyboardEvent);
+        expect(registrationManagerMock.prototype.invokeHandlers)
+            .toBeCalledWith(KeyEventType.KeyUp, "F1", keyboardEvent);
     });
 
     it("listens for keypress events and invokes handlers", () => {
         const keyboardEvent = new KeyboardEvent(
             KeyEventType.KeyPress, {
-            ctrlKey: false,
-            altKey: false,
-            key: "F1",
-        });
+                ctrlKey: false,
+                altKey: false,
+                key: "F1",
+            });
 
         window.dispatchEvent(keyboardEvent);
 
-        const registrationManagerMock = KeyboardRegistrationManager as jest.Mocked<typeof KeyboardRegistrationManager>;
-        expect(registrationManagerMock.prototype.invokeHandlers).toBeCalledWith(
-            KeyEventType.KeyPress, "F1", keyboardEvent);
+        expect(registrationManagerMock.prototype.invokeHandlers)
+            .toBeCalledWith(KeyEventType.KeyPress, "F1", keyboardEvent);
+    });
+
+    describe("Disabling keyboard events for input form elements", () => {
+        it("ignores keypress events when UI is focused on an input element", () => {
+            const keyboardEvent = new KeyboardEvent(
+                KeyEventType.KeyPress, {
+                    ctrlKey: false,
+                    altKey: false,
+                    key: "a",
+                });
+
+            Object.defineProperty(document, "activeElement", {
+                get: jest.fn(() => document.createElement("input")),
+                configurable: true,
+            });
+
+            window.dispatchEvent(keyboardEvent);
+
+            expect(registrationManagerMock.prototype.invokeHandlers).not.toBeCalled();
+        });
+
+        it("ignores keypress events when UI is focused on an textarea element", () => {
```

Comment:
Nit: textarea should be 2 words

Reply:
An HTML `textarea` is one word - keeping as is.